include/tester/db/Database.hpp

```diff
@@ -1,10 +1,203 @@
+    //         result_array << creation_time_doc;
+    //     }
+    //     // return results;
+    //     return bsoncxx::document::value{bsoncxx::builder::stream::open_array << result_array << bsoncxx::builder::stream::close_array << bsoncxx::builder::stream::finalize};
+    // }
+
+    // std::vector<bsoncxx::document::value> get_all_headers(const std::string& collection_name)
+    // {
+    //     mongocxx::collection collection = _database[collection_name];
+    //     mongocxx::cursor cursor = collection.find({});
+
+    //     std::vector<bsoncxx::document::value> results;
+    //     bsoncxx::builder::stream::array result{};
+    //     for (auto&& doc : cursor) {
+    //         result << bsoncxx::builder::stream::open_document;
+    //         if(doc["_id"])
+    //         {
+    //             // std::cout << doc["_id"].get_oid().value.to_string() <<std::endl;
+    //             result << "_id" << doc["_id"].get_oid().value.to_string();
+    //         }
+
+    //         if(doc["creation-time"])
+    //         {
+    //             // std::cout << doc["creation-time"].get_string().value.to_string() <<std::endl;
+    //             result << "creation-time" << doc["creation-time"].get_string().value.to_string();
+    //         }   
+            
+    //         result << bsoncxx::builder::stream::close_document;
+    //         // std::cout << bsoncxx::to_json(result.view());
+
+    //         // results.emplace_back(std::move(doc));
+
+    //     }
+
+    //     return result;
+    // }
+    bsoncxx::document::value getAllHeaders(const std::string& collection_name)
+    {
+        mongocxx::collection collection = _database[collection_name];
+        mongocxx::cursor cursor = collection.find({});
+
+        bsoncxx::builder::stream::document result{};
+        bsoncxx::builder::stream::array headers{};
+
+        for (auto&& doc : cursor) {
+            bsoncxx::builder::stream::document header_builder{};
+            header_builder << "_id" << doc["_id"].get_oid().value.to_string();
+
+            if(doc["creation-time"])
+            {
+                header_builder << "creation-time" << doc["creation-time"].get_string().value.to_string();
+            }
+
+            headers << header_builder;
+        }
+
+        result << "headers" << headers;
+
+        return result.extract();
+    }
+
+    std::vector<bsoncxx::document::value> get_all_documents(const std::string& collection_name)
+    {
+        mongocxx::collection collection = _database[collection_name];
+        mongocxx::cursor cursor = collection.find({});
+
+        std::vector<bsoncxx::document::value> results;
+        for (auto&& doc : cursor) {
+            results.emplace_back(std::move(doc));
+        }
+
+        return results;
+    }
+
+    bsoncxx::stdx::optional<std::string> get_field(const std::string& collection_name, const std::string& id, const std::string& field_name)
```

Comment:
`id` should be passed by value and be used with `std::move`

Reply:
done